requre/import_system.py

```diff
@@ -169,3 +69,144 @@ def revert_import_system():
     :return: None
     """
     builtins.__import__ = ORIGIN_IMPORT
+
+
+class UpgradeImportSystem:
+    def __init__(self, filters, debug_file: Optional[str] = None) -> None:
+        self.filters = filters
+        self.debug_file = debug_file
+        self._original_import = None
+        self.reverse: List[Tuple[Any, Any, Any]] = []
```

Comment:
I've used for similar reason that ``replace_dict: dict = {}``
so please merge both of them tohether.
that replace dict, is remembering what were replaced, and avoid multiple replacing, because python will reuse the module if already imported.
There is big magic around that :-)

Reply:
I've created a class for saving replacements.